### PR TITLE
Don't load cuda unless explicitly enabled

### LINKED ISSUE #2631: CUDA tries to turn on in C++03 mode
My workstation has CUDA on it and the library correctly detects it. However, If I try to compile the library with `DEAL_II_CXX_VERSION_FLAG="-std=c++03"` and do not explicitly request CUDA, then CMake fails with a complaint that CUDA does not work without C++11 support. Put another way: I have to explicitly disable CUDA if I want to compile in C++03 mode.

I believe our usual strategy here is to only fail if the user requests feature support that doesn't work, i.e., split the logic between `DEAL_II_HAVE_CUDA` and `DEAL_II_WITH_CUDA`, and not fail if the feature cannot be used and it was not requested.

### PR DESCRIPTION
**Pull request by kalj**
Fixes #2631 .

### CODE REVIEW

**Comment by drwells** [comment]
This looks like it should fix things. Thank you for addressing this quickly :)

Question about CUDA, since I don't know anything about it: The other external libraries are set up with `DEAL_II_PACKAGE_HANDLE`. Does this not make sense with CUDA?

**Comment by kalj** [comment]
@drwells It seems that most of the other features have customized FindPACKAGE.cmake files bundled with Deal.II, replacing the original files shipped with CMake. We have not yet had reason to do so for CUDA.

File: cmake/configure/configure_1_cuda.cmake [hunk 1]
```diff
@@ -19,26 +19,29 @@
...
-    IF(NOT DEAL_II_WITH_CXX11)
-      MESSAGE(FATAL_ERROR "\n"
-        "CUDA only supported with C++11. Reconfigure with DEAL_II_WITH_CXX11=ON.\n"
-        )
-    ENDIF()
+      IF(NOT DEAL_II_WITH_CXX11)
+        MESSAGE(FATAL_ERROR "\n"
+          "CUDA only supported with C++11. Reconfigure with DEAL_II_WITH_CXX11=ON.\n"
+          )
+      ENDIF()
```
Comment (tamiko) [inline cmake/configure/configure_1_cuda.cmake]: As a general guideline. Please, don't put a FATAL_ERROR message into the configuration stage :-)

**Comment by tamiko** [comment]
Superseded by #2639.

**Comment by kalj** [comment]
@tamiko Thanks for the feedback. :)

**Pull request closed without merging**